### PR TITLE
Merge dev into master

### PR DESCRIPTION
**Pull request by rseibane**
## Overview

Update master branch as we've been publishing new updates to `pub.dev` but not updating the branch.

- Add web support.
- Update entities to reflect the latest properties: bitcoinUnit, publicProfile...
- Add endpoint to get the exchange rate.
- Migrate to null-safety.
- Use query parameters instead of body in GET endpoints.

### CODE REVIEW

File: lib/src/models/cloud_api_exception.dart [hunk 1]
```diff
@@ -5,38 +5,40 @@ part 'cloud_api_exception.freezed.dart';
 @freezed
 abstract class CloudApiException with _$CloudApiException implements Exception {
   const factory CloudApiException.invalidRequest({
-    String message,
+    String? message,
```
Comment (rseibane) [inline lib/src/models/cloud_api_exception.dart:8]: Same as in core. This can be empty but not null.

File: lib/src/models/exchange_rate.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'exchange_rate.g.dart';
+
+@JsonSerializable()
+class ExchangeRate {
+  final String? fiatSymbol;
```
Comment (rseibane) [inline lib/src/models/exchange_rate.dart:7]: I don't think this can come as null

File: lib/src/models/payment_parameters.dart [hunk 3]
```diff
@@ -6,10 +6,10 @@ part 'payment_parameters.g.dart';
 
 @JsonSerializable(includeIfNull: false)
 class PaymentParameters {
-  final String description;
-  final String appAction;
-  final List<PaymentRequestItem> receivers;
-  final Attachment attachment;
+  final String? description;
```
Comment (rseibane) [inline lib/src/models/payment_parameters.dart:9]: Same here. Not sure about what happens with attachment but not nulls for sure

File: lib/src/models/payment_request_item.dart [hunk 4]
```diff
@@ -4,21 +4,21 @@ part 'payment_request_item.g.dart';
 
 @JsonSerializable()
 class PaymentRequestItem {
-  /// Receiver handler without $ or paymail 
-  final String destination;
+  /// Receiver handler without $ or paymail
```
Comment (rseibane) [inline lib/src/models/payment_request_item.dart:7]: Another one

File: lib/src/models/payment_result.dart [hunk 5]
```diff
@@ -9,19 +9,19 @@ part 'payment_result.g.dart';
...
 class PaymentResult {
-  final String transactionId;
-  final String note;
-  final String appAction;
-  final int time;
-  final PaymentType type;
-  final double satoshiFees;
-  final double satoshiAmount;
-  final double fiatExchangeRate;
+  final String? transactionId;
```
Comment (rseibane) [inline lib/src/models/payment_result.dart:12]: Same here

File: lib/src/models/private_profile.dart [hunk 6]
```diff
@@ -1,16 +1,19 @@
-
 import 'package:json_annotation/json_annotation.dart';
 
 part 'private_profile.g.dart';
 
 @JsonSerializable()
 class UserPrivateProfile {
-  final String phoneNumber;
-  final String email;
+  final String? phoneNumber;
```
Comment (rseibane) [inline lib/src/models/private_profile.dart:7]: I think they can come empty but not null.

File: lib/src/models/spendable_balance.dart [hunk 7]
```diff
@@ -4,17 +4,18 @@ part 'spendable_balance.g.dart';
 
 @JsonSerializable()
 class SpendableBalance {
-  final double spendableSatoshiBalance;
-  final double spendableFiatBalance;
-  final String currencyCode;
+  final double? spendableSatoshiBalance;
```
Comment (rseibane) [inline lib/src/models/spendable_balance.dart:7]: Not null for sure

File: lib/src/models/transaction_participant.dart [hunk 8]
```diff
@@ -4,10 +4,10 @@ part 'transaction_participant.g.dart';
 
 @JsonSerializable()
 class TransactionParticipant {
-  final String type;
-  final String alias;
-  final String displayName;
-  final String profilePictureUrl;
+  final String? type;
```
Comment (rseibane) [inline lib/src/models/transaction_participant.dart:7]: Some fields can be empty but not null

File: lib/src/sdk/handcash_connect.dart [hunk 9]
```diff
@@ -3,12 +3,14 @@ import 'package:handcash_connect_sdk/src/sdk/handcash_cloud_account.dart';
 import 'package:handcash_connect_sdk/src/services/service_factory.dart';
 
 class HandCashConnect {
-  static String _appId;
-  static Environment _environment;
+  static String? _appId;
+  static Environment? _environment;
```
Comment (rseibane) [inline lib/src/sdk/handcash_connect.dart:7]: Can this be null?

File: lib/src/sdk/profile.dart [hunk 10]
```diff
@@ -1,31 +1,32 @@
...
-import 'package:handcash_connect_sdk/src/services/profile/entities/handle_request.dart';
 import 'package:handcash_connect_sdk/src/services/profile/handcash_profile_service.dart';
 
 class Profile {
   final HandCashProfileService _profileService;
 
   Profile(this._profileService);
 
-  Future<UserProfile> getCurrentProfile() async {
+  Future<UserProfile?> getCurrentProfile() async {
```
Comment (rseibane) [inline lib/src/sdk/profile.dart:11]: That shouldn't return nulls. An object or throw an error

File: lib/src/sdk/wallet.dart [hunk 11]
```diff
@@ -1,29 +1,33 @@
...
+import 'package:handcash_connect_sdk/src/models/exchange_rate.dart';
 import 'package:handcash_connect_sdk/src/services/wallet/handcash_wallet_service.dart';
 
 class Wallet {
   final HandCashWalletService _walletService;
 
   Wallet(this._walletService);
 
-  Future<SpendableBalance> getSpendableBalance(String currencyCode) async {
+  Future<SpendableBalance?> getSpendableBalance(String currencyCode) async {
```
Comment (rseibane) [inline lib/src/sdk/wallet.dart:12]: Same here

File: lib/src/services/handcash_auth_interceptor.dart [hunk 12]
```diff
@@ -7,7 +7,7 @@ import 'package:handcash_connect_sdk/src/services/authentication_hash_signer.dar
 import 'package:tuple/tuple.dart';
 
 class HandCashAuthInterceptor extends RequestInterceptor {
-  final String authToken;
+  final String? authToken;
```
Comment (rseibane) [inline lib/src/services/handcash_auth_interceptor.dart:10]: This should be required

File: lib/src/services/handcash_connect_api_error.dart [hunk 13]
```diff
@@ -1,9 +1,9 @@
 import 'dart:convert';
 
 class HandCashConnectApiError implements Exception {
-  final int httpStatusCode;
-  final String message;
-  final Object info;
+  final int? httpStatusCode;
```
Comment (rseibane) [inline lib/src/services/handcash_connect_api_error.dart:4]: Not sure about info but statusCode and message shouldn't be null

File: lib/src/services/profile/entities/handle_request.dart [hunk 14]
```diff
@@ -4,11 +4,12 @@ part 'handle_request.g.dart';
 
 @JsonSerializable()
 class HandleRequest {
-  final List<String> aliases;
+  final List<String>? aliases;
```
Comment (rseibane) [inline lib/src/services/profile/entities/handle_request.dart:7]: This should be required

File: lib/src/services/wallet/entities/spendable_balance_request.dart [hunk 15]
```diff
@@ -4,11 +4,12 @@ part 'spendable_balance_request.g.dart';
 
 @JsonSerializable()
 class SpendableBalanceRequest {
-  final String currencyCode;
+  final String? currencyCode;
```
Comment (rseibane) [inline lib/src/services/wallet/entities/spendable_balance_request.dart:7]: Unused now that we move to query parameters rather than body?

**Review by rseibane — COMMENTED** [review]
Commented on things to adapt after migrating to null-safety @alfredo-handcash

File: lib/src/sdk/handcash_connect.dart (hunk 9 above)
Comment (alfredo-handcash) [reply lib/src/sdk/handcash_connect.dart:7]: Yeah, it´s null until uninitialized

File: lib/src/services/wallet/entities/spendable_balance_request.dart (hunk 15 above)
Comment (alfredo-handcash) [reply lib/src/services/wallet/entities/spendable_balance_request.dart:7]: Yes, I removed it!

File: lib/src/sdk/handcash_connect.dart (hunk 9 above)
Comment (rseibane) [reply lib/src/sdk/handcash_connect.dart:7]: Can we use `late` in this case?

**Review by alfredo-handcash — APPROVED** [review]
LGTM!

**Pull request merged**